docarray/index/backends/mongodb_atlas.py

```diff
@@ -0,0 +1,513 @@
+        :param query: the query to execute
+        :param args: positional arguments to pass to the query
+        :param kwargs: keyword arguments to pass to the query
+        :return: the result of the query
+        """
+        ...
+
+    @dataclass
+    class DBConfig(BaseDocIndex.DBConfig):
+        mongo_connection_uri: str = 'localhost'
+        index_name: Optional[str] = None
+        database_name: Optional[str] = "db"
+        default_column_config: Dict[Type, Dict[str, Any]] = field(
+            default_factory=lambda: defaultdict(
+                dict,
+                {
+                    bson.BSONARR: {
+                        'distance': 'COSINE',
+                        'oversample_factor': OVERSAMPLING_FACTOR,
+                        'max_candidates': MAX_CANDIDATES,
+                        'indexed': False,
+                        'index_name': None,
+                        'penalty': 1,
+                    },
+                    bson.BSONSTR: {
+                        'indexed': False,
+                        'index_name': None,
+                        'operator': 'phrase',
+                        'penalty': 10,
+                    },
+                },
+            )
+        )
+
+    @dataclass
+    class RuntimeConfig(BaseDocIndex.RuntimeConfig):
+        ...
+
+    def python_type_to_db_type(self, python_type: Type) -> Any:
+        """Map python type to database type.
+        Takes any python type and returns the corresponding database column type.
+
+        :param python_type: a python type.
+        :return: the corresponding database column type,
+            or None if ``python_type`` is not supported.
+        """
+
+        type_map = {
+            int: bson.BSONNUM,
+            float: bson.BSONDEC,
+            collections.OrderedDict: bson.BSONOBJ,
+            str: bson.BSONSTR,
+            bytes: bson.BSONBIN,
+            dict: bson.BSONOBJ,
+            np.ndarray: bson.BSONARR,
+            AbstractTensor: bson.BSONARR,
+        }
+
+        for py_type, mongo_types in type_map.items():
+            if safe_issubclass(python_type, py_type):
+                return mongo_types
+        raise ValueError(f'Unsupported column type for {type(self)}: {python_type}')
+
+    def _doc_to_mongo(self, doc):
+        result = doc.copy()
+
+        for name in result:
+            if self._column_infos[name].db_type == bson.BSONARR:
+                result[name] = list(result[name])
+
+        result["_id"] = result.pop("id")
+        return result
+
+    def _docs_to_mongo(self, docs):
+        return [self._doc_to_mongo(doc) for doc in docs]
+
+    @staticmethod
+    def _mongo_to_doc(mongo_doc: dict) -> tuple[dict, float]:
+        result = mongo_doc.copy()
+        result["id"] = result.pop("_id")
+        score = result.pop("score", None)
+        return result, score
+
+    @staticmethod
+    def _mongo_to_docs(mongo_docs: Generator[Dict, None, None]) -> tuple[list[dict], list[float]]:
```

Comment:
`tuple` should be changed to `Tuple` from `typing`. This is causing tests to fail